CalibTracker/SiStripChannelGain/src/APVGainHelpers.cc

```diff
@@ -106,89 +106,91 @@ int APVGain::subdetectorPlane(const std::string& tag) {
-                     (subdetectorPlane==0 || subdetectorPlane==sPlane) &&
-                     (subdetectorSide==0  || subdetectorSide==sSide);
-
-        if (match) {
-            found.push_back((*it).monitor);
-        }
-        it++;
+    bool match = (subdetectorId==0 || subdetectorId==sId) && (subdetectorPlane==0 || subdetectorPlane==sPlane) && (subdetectorSide==0  || subdetectorSide==sSide);
+
+    if (match) {
+      found.push_back((*it).getMonitor());
+      LogDebug("APVGainHelpers")<<det_id<<" found: "<< tag << std::endl;
+      (*it).printAll();
     }
-    return found;
+    it++;
+  }
+  return found;
 }
 
 
 
 std::vector<std::pair<std::string,std::string>> 
 APVGain::monHnames(std::vector<std::string> VH, bool allPlanes, const char* tag) {
 
-    std::vector<std::pair<std::string,std::string>> out;
-    int re = (allPlanes)? 34 + VH.size() : VH.size();
-    out.reserve( re );
-
-
-    std::string Tag = tag;
-    if (Tag.length())  Tag = "__" + Tag;
-
-    std::string h_tag = "";
-    std::string h_tit = "";
-
-    if (allPlanes) {
-        // Names of monitoring histogram for TIB layers
-        int TIBlayers = 4;  //number of TIB layers.
-        for(int i=1; i<=TIBlayers; i++) {
-            h_tag = "TIB_layer_" + std::to_string(i) + Tag;
-            h_tit = h_tag; std::replace(h_tit.begin(),h_tit.end(),'_',' ');
-            out.push_back(std::pair<std::string,std::string>(h_tag,h_tit));
-        }
-        // Names of monitoring histogram for TOB layers
-        int TOBlayers = 6;  //number of TOB layers
-        for(int i=1; i<=TOBlayers; i++) {
-            h_tag = "TOB_layer_" + std::to_string(i) + Tag;
-            h_tit = h_tag; std::replace(h_tit.begin(),h_tit.end(),'_',' ');
-            out.push_back(std::pair<std::string,std::string>(h_tag,h_tit));
-        }
-        // Names of monitoring histogram for TID wheels
-        int TIDwheels = 3;  //number of TID wheels
-        for(int i=-TIDwheels; i<=TIDwheels; i++) {
-            if (i==0) continue;
-            if (i<0)  h_tag = "TIDminus_wheel_" + std::to_string(i) + Tag;
-            else      h_tag = "TIDplus_wheel_" + std::to_string(i) + Tag;
-            h_tit = h_tag; std::replace(h_tit.begin(),h_tit.end(),'_',' ');
-            out.push_back(std::pair<std::string,std::string>(h_tag,h_tit));
-        }
-        // Names of monitoring histogram for TEC wheels
-        int TECwheels = 9;  //number of TEC wheels
-        for(int i=-TECwheels; i<=TECwheels; i++) {
-            if (i==0) continue;
-            if (i<0) h_tag = "TECminus_wheel_" + std::to_string(i) + Tag;
-            else     h_tag = "TECplus_wheel_" + std::to_string(i) + Tag;
-            h_tit = h_tag; std::replace(h_tit.begin(),h_tit.end(),'_',' ');
-            out.push_back(std::pair<std::string,std::string>(h_tag,h_tit));
-        }
-    }
+  std::vector<std::pair<std::string,std::string>> out;
+  int re = (allPlanes)? 34 + VH.size() : VH.size();
```

Comment:
you are not introducing it here, but where does `34` come from?

Reply:
added a comment in the code, it's the total possible number of Strip layers+wheels.